internal/provisioner/plain/controllers/bundle_controller.go

```diff
@@ -359,7 +359,7 @@ func getObjects(bundleFS fs.FS) ([]client.Object, error) {
 	}
 	for _, e := range entries {
 		if e.IsDir() {
-			continue
+			return nil, fmt.Errorf("directory is not allowed: %q", e.Name())
```

Comment:
nit: should this error string be referring to subdirectories to be more clear? for example `fmt.Errorf("subdirectories are not allowed in the bundle image filesystem: found %q", e.Name())`

Reply:
It's a clearer message.  I'll update.  Thanks!